WordPress/gutenberg
# Background image: add support for absolute theme path URLs

Moved to https://github.com/WordPress/gutenberg/pull/61401

> [!NOTE]
> This PR is experimental and will be a reference for ongoing work. See https://github.com/WordPress/gutenberg/pull/60578#issuecomment-2087925879

## What? And How??!!!

Part of:

- https://github.com/WordPress/gutenberg/issues/54336

This PR proposes a way to add support for theme relative url paths for theme.json/templates/patterns by:

- for the frontend, checking for `"source": "theme"` on the background image object in background supports, and appending the current theme path. It uses `get_stylesheet_directory_uri` instead of `get_template_directory_uri` so that the path will reflect any active child theme.
- for the editor, adding a new property to editor settings `currentTheme`, that contains the current theme directory path. It also uses `get_stylesheet_directory_uri` instead of `get_template_directory_uri` so that the path will reflect any active child theme.


## Why?

So that themes don't have to use global styles and rather, insert theme-relative paths into templates, patterns and theme.json.

This enhances portability of themes as Gutenberg will use the theme's current directory, regardless of install.

## Testing Instructions

Assuming that `img/Untitled.jpg` exists in the theme dir, here's some test theme.json. Note only top-level background images are possible until https://github.com/WordPress/gutenberg/pull/60100 gets in.

```json
{
	"$schema": "https://schemas.wp.org/trunk/theme.json",
	"version": 2,
	"settings": {
		"appearanceTools": true
	},
	"styles": {
		"background": {
			"backgroundImage": {
				"source": "theme",
				"url": "img/Untitled.jpg"
			},
			"backgroundPosition": "center center",
			"backgroundRepeat": "no-repeat",
			"backgroundSize": "cover"
		}
	}
}
```

Here's a test pattern (could also be HTML). Add it under `/patterns` in the theme directory:

```php
<?php
/**
 * Title: Empty theme pattern
 * Slug: emptytheme/empty-pattern
 * Categories: featured
 */
?>

<div class="wp-block-group" style="min-height:299px"></div>

```

It should also work with templates. Add it under `/templates` in the theme directory:

```html

<div class="wp-block-group" style="min-height:500px">


<div class="wp-block-query">
	
	
	
	
</div>

</div>

```

1. For each approach above, check that the background image's absolute path is rendered into the `background-image` CSS value for both the frontend and editor.
2. Ensure to check both the site editor (global styles) and post editor (block supports).

### Testing Instructions for Keyboard
